### PR TITLE
feat: [PIE-3299]: cd failure strategy yaml

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

Default CD Failure Strategy API integration

##### Jira Links:

https://harness.atlassian.net/browse/PIE-3299

##### Screenshots:

https://user-images.githubusercontent.com/96711906/162201094-0e6bf4a9-7094-444e-bf09-5036f0bc0a1f.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by AshwinBhatkal-Harness** [comment]
retrigger jest

**Comment by AshwinBhatkal-Harness** [comment]
retrigger jest

**Review by bhavya-sinha — DISMISSED** [review]

**Comment by AshwinBhatkal-Harness** [comment]
retrigger jest

**Comment by AshwinBhatkal-Harness** [comment]
trigger cypress

**Comment by AshwinBhatkal-Harness** [comment]
retrigger jest

**Comment by AshwinBhatkal-Harness** [comment]
retrigger jest

**Comment by AshwinBhatkal-Harness** [comment]
trigger cypress

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 1]
```diff
@@ -119,6 +122,27 @@ export default function DeployStageSetupShell(): JSX.Element {
...
+          produce(data?.stage as StageElementConfig, draft => {
+            const jsonFromYaml = YAML.parse(defaultTo(stageYamlSnippet?.data, '')) as StageElementConfig
+            if (!draft.failureStrategies) {
+              draft.failureStrategies = jsonFromYaml.failureStrategies
+            }
+          })
+        )
+      }
+    }
+  }, [stageYamlSnippet])
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:144]: Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'data?.stage', 'loading', 'refetch', and 'updateStage'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232284)
```suggestion
  }, [data?.stage, loading, refetch, stageYamlSnippet, updateStage])
```

**Review by Tamaraius — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx (hunk 1 above)
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:144]: Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'data?.stage', 'loading', 'refetch', and 'updateStage'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232284)
```suggestion
  }, [data?.stage, loading, refetch, stageYamlSnippet, updateStage])
```

**Pull request closed without merging**